src/Microsoft.NetCore.Analyzers/CSharp/Runtime/CSharpUseOrdinalStringComparison.Fixer.cs

```diff
@@ -47,11 +47,10 @@ protected override bool IsInIdentifierNameContext(SyntaxNode node)
 
         protected override async Task<Document> FixIdentifierName(Document document, SyntaxGenerator generator, SyntaxNode root, SyntaxNode identifier, CancellationToken cancellationToken)
         {
-            InvocationExpressionSyntax invokeParent = identifier?.Parent?.FirstAncestorOrSelf<InvocationExpressionSyntax>();
-            if (invokeParent != null)
+            if (identifier?.Parent?.FirstAncestorOrSelf<InvocationExpressionSyntax>() is InvocationExpressionSyntax invokeParent)
             {
                 SemanticModel model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                if (model.GetSymbolInfo((IdentifierNameSyntax)identifier, cancellationToken).Symbol is IMethodSymbol methodSymbol && CanAddStringComparison(methodSymbol, model))
+                if (model.GetSymbolInfo((IdentifierNameSyntax)identifier!, cancellationToken).Symbol is IMethodSymbol methodSymbol && CanAddStringComparison(methodSymbol, model))
```

Comment:
📝 This suppression is not supposed to be needed. If it's not compiling for you with this removed, a compiler bug should be filed.